File: 2D_sym_python/plants/Sow_thistle.py
from Plant import*

class Sow_thistle(Plant):

    def __init__(self):
        super().__init__()
        self.name='s'
    
    def action(self, world):
        for i in range(3):
            self.defaultAction(world)
    
    def GetName(self):
        return 's'
    
    def GetOrganismFullName(self):
        return "Sow_thistle"

File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Sow_thistle.cpp
#include <iostream>
#include "Sow_thistle.h"

Sow_thistle::Sow_thistle() {
	this->name = 's';
}

void Sow_thistle::action(World* world) {
	for (int i = 0; i < 3; i++) {
		this->defaultAction(world);
	}
}

char Sow_thistle::GetName() const {
	return 's';
}

string Sow_thistle::GetOrganismFullName() const {
	return "Sow_thistle";
}

Sow_thistle::~Sow_thistle() {

}